crepecake/src/main/java/com/google/cloud/tools/crepecake/http/Connection.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.http;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.URL;
+
+/** Sends an HTTP {@link Request} and stores the {@link Response}. */
+public class Connection implements Closeable {
+
+  private static final HttpRequestFactory HTTP_REQUEST_FACTORY =
+      new NetHttpTransport().createRequestFactory();
+
+  private final HttpRequestFactory requestFactory;
+
+  private HttpResponse httpResponse;
+
+  /** The URL to send the request to. */
+  private GenericUrl url;
+
+  /**
+   * Make sure to wrap with a try-with-resource to ensure that the connection is closed after usage.
+   */
+  public Connection(URL url) throws IOException {
+    this(url, HTTP_REQUEST_FACTORY);
+  }
+
+  @VisibleForTesting
+  Connection(URL url, HttpRequestFactory requestFactory) {
+    this.url = new GenericUrl(url);
+    this.requestFactory = requestFactory;
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (httpResponse == null) {
+      return;
+    }
+
+    httpResponse.disconnect();
+  }
+
+  /** Sends the request with method GET. */
+  public Response get(Request request) throws IOException {
```

Comment:
Given that the request factory is essentially a constant, it feels like this can be much simplified with static get and post methods that take a request and a URL.

Reply:
The reason these aren't in static methods right now is because I wanted to provide the closeable interface so that the connection can be auto-closed with the try-with-resource block.